js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,372 @@
+import Button from '../../common/components/Button';
+import LinkButton from '../../common/components/LinkButton';
+import Page from '../../common/components/Page';
+import Select from '../../common/components/Select';
+import Switch from '../../common/components/Switch';
+import icon from '../../common/helpers/icon';
+import listItems from '../../common/helpers/listItems';
+import ItemList from '../../common/utils/ItemList';
+import Stream from '../../common/utils/Stream';
+import LoadingModal from './LoadingModal';
+import ExtensionPermissionGrid from './ExtensionPermissionGrid';
+import saveSettings from '../utils/saveSettings';
+import ExtensionData from '../utils/ExtensionData';
+import extensionEnabled from '../utils/extensionEnabled';
+
+export default class ExtensionPage extends Page {
+  oninit(vnode) {
+    super.oninit(vnode);
+
+    this.loading = false;
+    this.extension = app.data.extensions[this.attrs.id];
+    this.changingState = false;
+    this.settings = {};
+
+    this.infoFields = {
+      discuss: 'fas fa-comment-alt',
+      documentation: 'fas fa-book',
+      support: 'fas fa-life-ring',
+      website: 'fas fa-link',
+      donate: 'fas fa-donate',
+    };
+
+    // Backwards compatibility layer will be removed in
+    // Beta 16
+    if (app.extensionSettings[this.extension.id]) {
+      app.extensionData[this.extension.id] = app.extensionSettings[this.extension.id];
+    }
+  }
+
+  className() {
+    return this.extension.id + '-Page';
+  }
+
+  view() {
+    return (
+      <div className={'ExtensionPage ' + this.className()}>
+        {this.header()}
+        {!this.isEnabled() ? (
+          <div className="container">
+            <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.enable_to_see')}</h2>
+          </div>
+        ) : (
+          <div className="ExtensionPage-body">{this.sections().toArray()}</div>
+        )}
+      </div>
+    );
+  }
+
+  header() {
+    return [
+      <div className="ExtensionPage-header">
+        <div className="container">
+          <div className="ExtensionTitle">
+            <span className="ExtensionIcon" style={this.extension.icon}>
+              {this.extension.icon ? icon(this.extension.icon.name) : ''}
+            </span>
+            <div className="ExtensionName">
+              <label>{this.extension.extra['flarum-extension'].title}</label>
+            </div>
+            <div className="ExtensionPage-headerTopItems">
+              <ul>{listItems(this.topItems().toArray())}</ul>
+            </div>
+          </div>
+          <div className="helpText">{this.extension.description}</div>
```

Comment:
Should this also be changed to AdminHeader-helpText?

Reply:
No, because it's not apart of the Admin header, this is technically the extension header